flair/data.py

```diff
@@ -357,19 +357,138 @@ def __repr__(self) -> str:
+    https://github.com/fnl/segtok
+    """
+    tokens: List[Token] = []
+
+    words: List[str] = []
+    sentences = split_single(text)
+    for sentence in sentences:
+        contractions = split_contractions(word_tokenizer(sentence))
+        words.extend(contractions)
+
+    # determine offsets for whitespace_after field
+    index = text.index
+    current_offset = 0
+    previous_word_offset = -1
+    previous_token = None
+    for word in words:
+        try:
+            word_offset = index(word, current_offset)
+            start_position = word_offset
+        except:
+            word_offset = previous_word_offset + 1
+            start_position = (
+                current_offset + 1 if current_offset > 0 else current_offset
+            )
+
+        token = Token(text=word, start_position=start_position, whitespace_after=True)
+        tokens.append(token)
+
+        if (previous_token is not None) and word_offset - 1 == previous_word_offset:
+            previous_token.whitespace_after = False
+
+        current_offset = word_offset + len(word)
+        previous_word_offset = current_offset - 1
+        previous_token = token
+
+    return tokens
+
+
+def build_spacy_tokenizer(model) -> Callable[[str], List[Token]]:
+    """
+    Wrap Spacy model to build a tokenizer for the Sentence class.
+    :param model a Spacy V2 model
+    :return a tokenizer function to provide to Sentence class constructor
+    """
+    try:
+        from spacy.language import Language
+        from spacy.tokens.doc import Doc
+        from spacy.tokens.token import Token as SpacyToken
+    except ImportError:
+        raise ImportError(
+            "Please install Spacy v2.0 or better before using the Spacy tokenizer, otherwise you can use segtok_tokenizer as advanced tokenizer."
+        )
+
+    model: Language = model
+
+    def tokenizer(text: str) -> List[Token]:
+        doc: Doc = model.make_doc(text)
+        previous_token = None
+        tokens: List[Token] = []
+        for word in doc:
+            word: SpacyToken = word
+            token = Token(
+                text=word.text, start_position=word.idx, whitespace_after=True
+            )
+            tokens.append(token)
+
+            if (previous_token is not None) and (
+                token.start_pos - 1
+                == previous_token.start_pos + len(previous_token.text)
+            ):
+                previous_token.whitespace_after = False
+
+            previous_token = token
+        return tokens
+
+    return tokenizer
+
+
 class Sentence(DataPoint):
     """
-    A Sentence is a list of Tokens and is used to represent a sentence or text fragment.
+       A Sentence is a list of Tokens and is used to represent a sentence or text fragment.
     """
 
     def __init__(
         self,
         text: str = None,
-        use_tokenizer: bool = False,
+        tokenizer: Callable[[str], List[Token]] = space_tokenizer,
```

Comment:
This breaks backwards compatibility to downstream code that uses Flair with earlier versions. Perhaps the signature could be changed to: 
```python
    def __init__(
        self,
        text: str = None,
        use_tokenizer: Union[bool, Callable[[str], List[Token]]] = space_tokenizer,
        labels: Union[List[Label], List[str]] = None,
        language_code: str = None,
    ):
```

i.e. call it `use_tokenizer` instead of `tokenizer` and allow passing a bool in addition to callable. Then, a few lines down one could add: 

```python
        tokenizer = use_tokenizer
        if type(use_tokenizer) == bool:
            tokenizer = segtok_tokenizer if use_tokenizer else space_tokenizer
```

i.e. by default if a callable is passed `tokenizer = use_tokenizer` but if a bool is passed the tokenizer is instead initialized with the earlier behavior, i.e. `tokenizer = segtok_tokenizer` if `use_tokenizer=True`.